### PR TITLE
Add no-pkg-flags-to-cmj test 

### PR DESCRIPTION
**Pull request by anmonteiro**
Re-hash of #386

### CODE REVIEW

File: test/no-package-flags-to-cmj.t [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  $ melc -bs-stop-after-cmj lib/a.ml -o lib/.objs/melange/a.cmj
+
+  $ melc -I lib/.objs/melange -bs-stop-after-cmj app/b.ml -o app/.objs/melange/b.cmj
+
+Linking step
+
+  $ melc -bs-package-output lib/ lib/.objs/melange/a.cmj -o output/lib/a.js
+
+If we pass some file extension or module system in `-bs-package-output`, it
```
Comment (anmonteiro) [inline test/no-package-flags-to-cmj.t:24]: @jchavarri I actually changed this. If you pass `bs-package-output` with package specs Melange will use that. The correct flag to use to control JS output at this stage is `-bs-module-type`

Comment (jchavarri) [reply test/no-package-flags-to-cmj.t:24]: Sounds good.

re: passing `-o` with file extension and `-bs-package-output`, I am not sure if we should fail? It's a bit confusing that user can pass file extension in two places.

Comment (anmonteiro) [reply test/no-package-flags-to-cmj.t:24]: it's indeed a bit confusing, but I think we don't need to care about that:
- the CLI interface is for "expert" usage
- people will mostly interact with melange through `mel` or `dune`

Comment (anmonteiro) [reply test/no-package-flags-to-cmj.t:24]: I think that's one (maybe last?) thing to improve on later. I think the code is a lot cleaner after #384, as there's now a separation between the "package specs" (as recorded in the `.cmj` -- what's the relative path of this module?) and the output information we'll need for emission ("what module system and JS suffix should this file have?").

File: test/no-package-flags-to-cmj.t [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  $ melc -bs-package-output app/ -I lib/.objs/melange app/.objs/melange/b.cmj -o output/app/b.js
+  File "_none_", line 1:
+  Error: a.js not found, needed in script mode 
+  [2]
+
+Passing `-I output/lib` fixes it
+
+  $ melc -bs-package-output app/ -I lib/.objs/melange -I output/lib app/.objs/melange/b.cmj -o output/app/b.js
+
+B depends on A, so it should import a.js in the right path
```
Comment (jchavarri) [inline test/no-package-flags-to-cmj.t:55]:
```suggestion
B depends on A, so it should import a.js in the right path, the default module system used is commonJS
```

**Review by jchavarri — APPROVED** [review]
I only took a quick look, but the internal split between batch and separate compilation sounds like a great idea.

**Pull request merged**